service/matching/fx.go

```diff
@@ -92,18 +99,34 @@ func PersistenceMaxQpsProvider(
 	return service.PersistenceMaxQpsFn(serviceConfig.PersistenceMaxQPS, serviceConfig.PersistenceGlobalMaxQPS)
 }
 
-func ServiceProvider(
-	serviceResource resource.Resource,
-	grpcServerOptions []grpc.ServerOption,
-	serviceConfig *Config,
-) *Service {
-	return &Service{
-		Resource: serviceResource,
-		status:   common.DaemonStatusInitialized,
-		config:   serviceConfig,
-		server:   grpc.NewServer(grpcServerOptions...),
-		handler:  NewHandler(serviceResource, serviceConfig),
-	}
+func ServiceResolverProvider(membershipMonitor membership.Monitor) (membership.ServiceResolver, error) {
+	return membershipMonitor.GetResolver(common.MatchingServiceName)
+}
+
+func HandlerProvider(
```

Comment:
use NewHandler and avoid this HandlerProvider?

Reply:
This is used to resolve some named dependencies.